WordPress/gutenberg
# docgen: don't transpile the package code

I noticed that the `docgen` package is Node.js only and exports only a bin executable (not even a module API), and yet it's transpiled by Babel into `build` and `build-module` folders.

This PR fixes that by renaming the `src` module to `lib` so that the build scripts don't pick the sources automatically when they search with a `src/**/*.js` glob. The `test` directory is no longer a subfolder of `lib`, but its sibling at the top level.

The `package/env` is a good prior art here: it uses the `lib` folder already.

**How to test:**
Run `npm run docs:build` and check that it succeeds. Also run the `docgen` unit tests. There should be no observable changes in the package behavior at all.
